EasyLM/models/llama/llama_train_ppo.py

```diff
@@ -584,34 +585,30 @@ def save_checkpoint(policy_train_state, value_train_state, step, milestone=False
                 del value_params
 
         # Load reference
-        reference_train_state, reference_params = None, None
+        reference_params = None
         if FLAGS.load_checkpoint_policy != '':
             print("Loading checkpoint (reference) ... (may take time to download)")
             reference_train_state, reference_params = checkpointer.load_trainstate_checkpoint(FLAGS.load_checkpoint_policy, train_state_shapes_policy, shard_fns_policy)
             print("Checkpoint (reference) loaded.")
-        if reference_train_state is None:
-            if reference_params is None:
-                reference_train_state = sharded_init_fn_policy(next_rng())
-            else:
-                if not FLAGS.use_tpu:
-                    reference_params = flax.core.frozen_dict.unfreeze(reference_params)
-                reference_train_state = sharded_create_trainstate_from_params_policy(reference_params)
-                del reference_params
+            assert reference_train_state is None
+        if reference_params is None:
+            reference_params = copy.deepcopy(policy_train_state.params)
```

Comment:
does this work correctly on tpu?

Reply:
Yep. I'm using this for my experiments now, seems no problem